ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_device_operation.cpp

```diff
@@ -155,21 +155,32 @@ BinaryDeviceOperation::shape_return_value_t BinaryDeviceOperation::compute_outpu
     const auto input_shape_a = tensor_args.input_tensor_a.tensor_attributes->shape;
     const auto input_shape_b = tensor_args.input_tensor_b.tensor_attributes->shape;
 
-    auto rank = std::max(input_shape_a.rank(), input_shape_b.rank());
+    const int rank_a = input_shape_a.rank();
+    const int rank_b = input_shape_b.rank();
+    const auto rank = std::max(rank_a, rank_b);
     std::vector<uint32_t> output_shape(rank, 0);
     std::vector<uint32_t> output_shape_with_tile_padding(rank, 0);
 
+    const SimpleShape logical_shape_a = input_shape_a.logical_shape();
+    const SimpleShape padded_shape_a = input_shape_a.padded_shape();
+
+    const SimpleShape logical_shape_b = input_shape_b.logical_shape();
+    const SimpleShape padded_shape_b = input_shape_b.padded_shape();
+
+    auto get_dim = [](int i, int rank, const auto& shape) {
+        return i + rank < rank ? shape[i] : 1;
```

Comment:
Something is wrong with this check. It's the same as `i < 0`. Are you sure? 

I see, that you've just moved the same logic, but it's clearly a bug.

Reply:
I think the intention was return abs(i) <= rank ? shape[i] : 1;

Reply:
Fixed

Reply:
The issue here is collision of operand_rank and larger_rank. I renamed variables and reorganized this code to make things clear.